sdk/python/generator/Generator.py

```diff
@@ -113,73 +94,59 @@ def bind_size_fields(struct_type):
 	for field in struct_type.layout:
 		if field.type_instance.is_array and isinstance(field.get_type().size, str):
 			size_field_name = field.get_type().size
-			# 'unfix' name when searching for associated 'size' field:
-			#  * fields are added using 'original' name,
-			#  * but 'size' property of given array type is already fixed
-			#
-			# (currently this only happens for metadata value)
-			if 'size_' == size_field_name:
-				size_field_name = 'size'
-
 			size_field = struct_type.get_field_by_name(size_field_name)
 			size_field.set_bound_field(field)
 
 		if field.type_instance.sizeof_value:
-			struct_field = struct_type.get_field_by_name(field.yaml_descriptor['value'])
+			struct_field = struct_type.get_field_by_name(field.ast_model.name[:-5])
```

Comment:
@gimre-xymcity - maybe you have some suggestion how to get around this hack? 😅

Reply:
I assume ast_model must have ".value" field (?), as it is present in yaml, e.g. (nem):
(although I'm somewhat wondering if "value" was intended name for this...)

```
 - comments: mosaic id size
    disposition: sizeof
    name: mosaic_id_size
    signedness: unsigned
    size: 4
    type: byte
    value: mosaic_id
```

Reply:
i looked again and was adding wrong field. 😅 - it works now.